src/Interpreters/executeSelectQuery.cpp

```diff
@@ -45,9 +45,15 @@ void executeQueryWithoutProcessor(DB::BlockInputStreamPtr & in, const std::funct
 
 namespace DB
 {
-void executeSelectQuery(const String & query, ContextPtr query_context, const std::function<void(Block &&)> & callback, bool internal)
+void executeSelectQuery(
+    const String & query,
+    ContextPtr query_context,
+    const std::function<void(Block &&)> & callback,
+    bool internal,
+    QueryProcessingStage::Enum stage,
```

Comment:
we shall have no cases we need pass in a different stage other than `Complete`, so we can drop this param. Same as `may_have_embedded_data` which shall be always `false`